internal/distro/distro.go

```diff
@@ -125,10 +125,11 @@ type ImageType interface {
 
 // The ImageOptions specify options for a specific image build
 type ImageOptions struct {
-	Size         uint64
-	OSTree       OSTreeImageOptions
-	Subscription *SubscriptionImageOptions
-	Facts        *FactsImageOptions
+	Size          uint64
+	OSTree        OSTreeImageOptions
+	Subscription  *SubscriptionImageOptions
+	Facts         *FactsImageOptions
+	RequiredSizes map[string]uint64
```

Comment:
Sorry, but I noticed this just now. 😕 

We tend to use `ImakeOptions` that are specific to the build and usually provided by users (except for `Facts`). I do not expect this to be the case for `RequiredSizes`. I thought that the `RequiredSizes` is supposed to be the image-type-specific property, therefore it would IMO make sense to make it part of the `imageType` struct instead.

Reply:
And consequently define the empty `map` in here https://github.com/osbuild/osbuild-composer/blob/4023b61e1a56a26f27388bb03724028ce25be2e2/internal/distro/fedora/distro.go#L154-L171 instead of the `iotRawImage()`.

Reply:
Agreed, and done.